### ISSUE TITLE
i18n support

### ISSUE THREAD

**Issue by tresf** [qzind/tray#273]
Pinging @liushuyu.

Have you done any i18n for Java?  I just spotted https://github.com/chunlinyao/tray/commit/f33e2750a3f5d374967855f48bebc48d3e6a3da9 which makes me think we could benefit from this.  Any interest?

**Comment by liushuyu** [comment]
Yes but that's a story from the long past. However, with some time I should be able to pick up the knowledge that's needed to accomplish the job.

**Comment by liushuyu** [comment]
... and BTW, which branch I should target for? 2.0 or 2.1?

**Comment by tresf** [comment]
@liushuyu let's target 2.1 for now.

**Comment by johann-sonntagbauer** [comment]
@liushuyu did you make any progress? I wanted to start the same maybe we can join forces :) 
I have some experience with java/swing development and can layout the foundation for the resource bundles and so forth.

**Comment by tresf** [comment]
@johann-sonntagbauer we have a working example ready in #290.  I've opened a PR on @liushuyu's behalf.

Note, we are exploiting the `.properties` file format by hacking in UTF-8 support.  We still need to write an ANT task to scrape all translatable strings out.  @liushuyu is still going to help with this but was delayed last we spoke.

**Comment by johann-sonntagbauer** [comment]
@tresf looks great so far :) anything I can support you or @liushuyu with?

**Comment by tresf** [comment]
He'll need to grant you commit access to push to that repo, but translation strings would be a good start.  There's a very basic example in `src/qz/common/resources/messages_zh_CN.properties` for starters.  If you have interest in the coding side too, an ANT task for scraping translation strings would be the next step.

**Comment by tresf** [comment]
A basic UI selector would be necessary as well.

**Comment by johann-sonntagbauer** [comment]
I will throw something together today in the evening. 
The Ant task should not be too hard, because the heavy lifting is already done. A RegularExpression extracting the gettext() calls would be sufficient. 

The UI selector is a little bit tricky. Do you plan to support dynamic locale selection? Or simply select the locale, store in properties file and restart application? Because for now there are some translated texts formulated as final static members.

**Comment by tresf** [comment]
Properties/restart option would be great.

**Comment by liushuyu** [comment]
Hi,

I am very sorry for the late response, I am not sure if I have 100% marked all the strings with `gettext` wrapper.

And I will try to finish the translation file  today or tomorrow.

Thanks

**Comment by johann-sonntagbauer** [comment]
@liushuyu no pressure. I am very thankfull for the hard work you all put into this project. I only want to give something back :)

**Comment by liushuyu** [comment]
>@liushuyu no pressure. I am very thankfull for the hard work you all put into this project. I only want to give something back :)

Thanks. 

And there's a hint you may have already seen in the translation file: you'll need to escape some symbols when scraping the source code like whitespaces.

**Comment by johann-sonntagbauer** [comment]
Short update. I am currently finalizing the locale selector. Ant task is in the making need to investigate little more how to integrate it best with qz builds. Problem for now is, that the Task is built with ant, but the built Task needs to be on the ant classpath :) 
 
@tresf what icon set does qz use? I am finalizing the locale selection and wanted to add some icons to the menu.

**Comment by tresf** [comment]
Our artwork is 100% original.  If you can provide a mock-up I can reach out to our artist.

**Comment by johann-sonntagbauer** [comment]
I am done with my changes. 
* UI for locale selection
* restart after locale change to proper update all static constants
* selected locale gets stored into settings properties
* mockup locale change icon
* ant task to extract text and store into properties files
* german translation

out of scope for now
* installer dialog for locale selection. (This is harder then I thought) 

How to proceed from here? I based my changes on top of the PR https://github.com/qzind/tray/pull/290
So two options: 
1. merge the two PRs into one
2. merge @liushuyu PR and I will rebase on top and formulate a seperate PR

**Comment by tresf** [comment]
> merge the two PRs into one

That's probably the best option.  @liushuyu or myself can pull your changes into the PR if you cross-link the commit(s) or the compare.

**Comment by johann-sonntagbauer** [comment]
@tresf ok I will formulate a PR based on @liushuyu changes so it will be easy to merge than.
